text/metrics/0146-metrics-prototype-scenarios.md

```diff
@@ -0,0 +1,220 @@
+a final application. It is a self-contained application which covers:
+
+* How to instrument the code in a vendor agnostic way
+* How to configure the SDK and exporter
+
+Considering there might be multiple grocery stores, the metrics we collect will
+have the store name as a dimension - which is fairly static (not changing while
+the store is running).
+
+The store has plenty supply of potato and tomato, with the following price:
+
+* Potato: $1.00 / ea
+* Tomato: $3.00 / ea
+
+Each customer has a unique name (e.g. customerA, customerB), a customer could
+come to the store multiple times. Here goes the Python snippet:
+
+```python
+store = GroceryStore("Portland")
+store.process_order("customerA", {"potato": 2, "tomato": 3})
+store.process_order("customerB", {"tomato": 10})
+store.process_order("customerC", {"potato": 2})
+store.process_order("customerA", {"tomato": 1})
+```
+
+When the store is closed, we will report the following metrics:
+
+### Order info
+
+| Store    | Customer  | Number of Orders | Amount (USD) |
+| -------- | --------- | ---------------- | ------------ |
+| Portland | customerA | 2                | 14.00        |
+| Portland | customerB | 1                | 30.00        |
+| Portland | customerC | 1                | 2.00         |
+
+### Items sold
+
+| Store    | Customer  | Item   | Count |
+| -------- | --------- | ------ | ----- |
+| Portland | customerA | potato | 2     |
+| Portland | customerA | tomato | 4     |
+| Portland | customerB | tomato | 10    |
+| Portland | customerC | potato | 2     |
+
+## Scenario 2: HTTP Server
+
+The _HTTP Server_ scenario covers how a library developer X could use metrics
+API to instrument a library, and how the application developer Y can configure
+the library to use OpenTelemetry SDK in a final application. X and Y are working
+for different companies and they don't communicate. The demo has two parts - the
+library (HTTP lib owned by X) and the server app (owned by Y):
+
+* How developer X could instrument the library code in a vendor agnostic way
+  * Performance is critical for X
+  * X doesn't know which metrics and which dimensions will Y pick
+  * X doesn't know the aggregation time window, nor the final destination of the
+    metrics
+* How developer Y could configure the SDK and exporter
+  * How should Y hook up the metrics SDK with the library
+  * How should Y configure the time window(s) and destination(s)
+  * How should Y pick the metrics and the dimensions
+
+### Library Requirements
+
+The library developer (developer X) will expose the following metrics out of
+box:
+
+### Pull Metrics
+
+These are pull metrics - the value is always available, and is only reported and
+collected based on the ask from consumer(s). If there is no ask from the
+consumer, the value will not be reported at all (e.g. there is no API call to
+fetch the room temperature unless someone is asking for the room temperature).
+
+#### Process CPU Usage
```

Comment:
Is "Process CPU Usage" really an HTTP server concern?   I feel like this may be a general instrumentation concern, and the HTTP metrics should really be focused only on things HTTP libraries do.  CPU consumption is more of a process-wide concern.

I'd suggest using `Active HTTP Connections` as the pull metric from the HTTP library.


For Server Room Temperature, I love what it's trying to do, but I'm having trouble buying it's part of an http library.  Maybe move it into its own instrumentation component?

Reply:
I've removed CPU since itself is a complex topic and not the focus on this OTEP/prototype.
I've extracted temperature/humidity to a separate lib to make the scenario more realistic/reasonable.

For Active HTTP Connections, I don't know if we want to do it by reporting "total received - total processed" or doing it differently (see the discussion [here](https://github.com/open-telemetry/opentelemetry-specification/pull/1378)). Given this discussion might take extra time, probably leave it out for now?

Reply:
The thing about CPU is that it's a perfect example of an asynchronous instrument in the draft API. We want these to be recorded though callbacks, because it's expensive. This means the value is returned in cumulative form, not in delta form.